File: src/wrapper/cs/classes/ConsoleLoggerProvider.cs
/*
 * Copyright (c) 2004-2019 Lightstreamer s.r.l., Via Campanini, 6 - 20124 Milano, Italy.
 * All rights reserved.
 * www.lightstreamer.com
 *
 * This software is the confidential and proprietary information of
 * Lightstreamer s.r.l.
 * You shall not disclose such Confidential Information and shall use it
 * only in accordance with the terms of the license agreement you entered
 * into with Lightstreamer s.r.l.
 */
namespace com.lightstreamer.log
{
  /// <summary>
  /// Simple concrete logging provider that logs on the system console.
  ///
  /// To be used, an instance of this class has to be passed to the library through the <seealso cref="com.lightstreamer.client.LightstreamerClient.setLoggerProvider(ILoggerProvider)"/>.
  /// </summary>
  public class ConsoleLoggerProvider: ILoggerProvider {
    readonly LSConsoleLoggerProvider _delegate;

    /// <summary>
    /// Creates an instace of the concrete system console logger.</summary>
    ///
    /// <param name="level"> The desired logging level. See <seealso cref="ConsoleLogLevel"/>.</param>
    public ConsoleLoggerProvider(int level) {
        this._delegate = new LSConsoleLoggerProvider(level);
    }

    public ILogger GetLogger(string category) {
        return _delegate.GetLogger(category);
    }
  }
}

File: src/wrapper/java/classes/com/lightstreamer/log/ConsoleLoggerProvider.java
/*
 * Copyright (C) 2023 Lightstreamer Srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.lightstreamer.log;

import com.lightstreamer.log.LSConsoleLoggerProvider;

/**
  Simple concrete logging provider that logs on the system console.
 
  To be used, an instance of this class has to be passed to the library through the {@link com.lightstreamer.client.LightstreamerClient#setLoggerProvider(LoggerProvider)}.
 */
public class ConsoleLoggerProvider implements LoggerProvider {
  final LSConsoleLoggerProvider delegate;

  /**
    Creates an instance of the concrete system console logger.
     
    @param level The desired logging level. See {@link ConsoleLogLevel}.
  */
  public ConsoleLoggerProvider(int level) {
    this.delegate = new LSConsoleLoggerProvider(level);
  }

  public Logger getLogger(String category) {
    return delegate.getLogger(category);
  }
}